crates/transaction-pool/src/pool/blob.rs

```diff
@@ -136,10 +127,30 @@ impl<T: PoolTransaction> BlobTransactions<T> {
         transactions
     }
 
+    /// Resorts the transactions in the pool based on the pool's current [PendingFees].
+    pub(crate) fn reprioritize(&mut self) {
+        let mut new_all_transactions: Vec<_> = Default::default();
+        while let Some(mut tx) = self.all.pop_last() {
+            tx.update_priority(&self.pending_fees);
+            new_all_transactions.push(tx);
+        }
+
+        // we need to update `by_id` as well because removal from `all` can only happen if the
+        // `BlobTransaction`s in each struct are consistent
+        for tx in self.by_id.values_mut() {
+            tx.update_priority(&self.pending_fees);
+        }
+
+        self.all = new_all_transactions.into_iter().collect();
```

Comment:
this does two allocations we could instead, do a mem::tak and push directly back to `all` so we don't need the temp vec here